mllib/src/test/scala/org/apache/spark/ml/util/MLTestingUtils.scala

```diff
@@ -92,21 +111,28 @@ object MLTestingUtils extends SparkFunSuite {
     types
       .map(t => t -> df.select(col(labelColName).cast(t), col(featuresColName)))
       .map { case (t, d) =>
-        t -> TreeTests.setMetadata(d, 0, labelColName).withColumn(censorColName, lit(0.0))
+        t -> TreeTests.setMetadata(d, 0, labelColName, featuresColName)
```

Comment:
minor: Not related to the changes in this PR, but I tend to prefer a single map statement like:

```scala
types.map { t =>
      val castDF = df.select(col(labelColName).cast(t), col(featuresColName))
      t -> TreeTests.setMetadata(castDF, 2, labelColName, featuresColName)
    }.toMap
```

This happens in the `genRegressionDFWithNumericLabelCol` as well.